### PR TITLE
Port event_rpcgen.py and check-dumpevents.py to Python 3.

### PR DESCRIPTION
**Pull request by Aerdan**
These scripts remain compatible with Python 2, and so the wrapper script has been modified to prefer Python 3 if available, then fall back to Python 2.

Relates to #558.

### CODE REVIEW

File: test/rpcgen_wrapper.sh [hunk 1]
```diff
@@ -25,19 +25,20 @@ exit_failed() {
...
-  PYTHON2=python
+  PYTHON=python
 fi
 
 srcdir=$1
 srcdir=${srcdir:-.}
 
-${PYTHON2} ${srcdir}/../event_rpcgen.py --quiet ${srcdir}/regress.rpc \
+${PYTHON} ${srcdir}/../event_rpcgen.py --quiet ${srcdir}/regress.rpc \
                test/regress.gen.h test/regress.gen.c
```
Comment (azat) [inline test/rpcgen_wrapper.sh]: Maybe just replace all this `if/elif` with just allowing use shebang? It will became cleaner in this case.

Comment (azat) [reply test/rpcgen_wrapper.sh]: Also this PR misses changes into `test/test.sh` that invokes `check-dumpevents.py` and to `CMakeLists.txt` which also invokes it, can you take a look?

**Comment by azat** [comment]
@Aerdan are you going to fix missing bits, or should I finish this?

**Comment by Aerdan** [comment]
I addressed the bits you requested I modify, yes.

**Comment by azat** [comment]
Didn't notice that, sorry (there were no notifications from github).
Anyway applied as 8b0aa7b36a3250fad4953f194c8a94ab25032583, thanks!

**Pull request closed without merging**